### PR TITLE
Creates first draft for CAP portal

### PR DESCRIPTION
**Pull request by mjudeikis**
This created an initial structure for Cloud Access Portal. At the moment binary still serves API, emulator and UI.

In addition, I moved to negroni as HTTP server as it has a lot of extensions, which we will need later. Like authentication middleware, handler management, etc. 

In the future, we might want to have API as a separate router if multiple services will be fronted with it. This will allow even better decoupling of the tooling.  

Next steps:
* Add some structure to UI (potentially vuejs and material UI) as they are leading players. + vuejs v3 will be typescript. 

In total this renames logviewer to the more generic name "SRE Dashboard" and we incorporate all into it as separate golang components.

### CODE REVIEW

**Comment by kargakis** [comment]
Defintely confusing

When I read CAP I think of https://en.wikipedia.org/wiki/CAP_theorem

**Comment by mjudeikis** [comment]
> When I read CAP I think of https://en.wikipedia.org/wiki/CAP_theorem

OCP https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle
We cant run away from this :D

**Comment by mjudeikis** [comment]
Dam How we need somebody with HTML, CSS, javascript skills here :dagger:

**Comment by mjudeikis** [comment]
This is missing vendor. This is why tests are faling, when LGTM comes I will commit vendor dir and merge

**Comment by kwoodson** [comment]
@mjudeikis 
any ideas on this?
```
error: could not run steps: could not wait for build: the build bin failed after 2m43s with reason DockerBuildFailed: Docker build strategy has failed.
```

/test images

**Comment by mjudeikis** [comment]
@kwoodson it was vendor. Didnt wanted to contaminate PR before the review. Added dependencies.

**Comment by kwoodson** [comment]
@mjudeikis did you want to squash as well?

Other than that, lgtm.

**Comment by mjudeikis** [comment]
done

File: cap/pkg/clusters/interface.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package clusters
+
+import "github.com/openshift/openshift-azure/pkg/api"
+
+type Cluster struct {
+	Name         string `json:"name,omitempty"`
```
Comment (jim-minter) [inline cap/pkg/clusters/interface.go]: it was a mistake of mine to add `json:"..."` tags here - they should be removed

File: cap/pkg/clusters/interface.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	Manifest     *api.OpenShiftManagedCluster
+	WorkspaceID  string `json:"workspaceId,omitempty"`
+	ROKubeconfig string
```
Comment (jim-minter) [inline cap/pkg/clusters/interface.go]: unused?

**Pull request merged**